core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2996,12 +2997,19 @@ else if (operation instanceof MergeInsert && caseColumnNames.isEmpty()) {
                     .map(fieldIndexes::get)
                     .collect(toImmutableList());
 
+            Set<ColumnHandle> nonNullableColumnHandles = metadata.getTableMetadata(session, targetTableHandle).getColumns().stream()
+                    .filter(column -> !column.isNullable() && !column.isHidden())
```

Comment:
Do we want to filter out hidden columns? What happens if someone assigns to a hidden column?

Reply:
I didn't know it was possible to assign to a hidden column.

I'll have to look at everywhere else where I've excluded hidden columns.